apis/metal3.io/v1alpha1/baremetalhost_types.go

```diff
@@ -740,6 +752,9 @@ type ProvisionStatus struct {
 
 	// The Bios set by the user
 	Firmware *FirmwareConfig `json:"firmware,omitempty"`
+
+	// Custom deploy procedure applied to the host.
+	CustomDeploy *CustomDeploy `json:"customDeploy,omitempty"`
```

Comment:
Not too much? We'd still have to deal with Image.URL possibly be "", CustomDeploy possibly be ""/`nil` and so on. A potential user confusion because of conflating too barely related terms may not be worth it.